server/src/internalClusterTest/java/org/opensearch/routing/SimpleRoutingIT.java

```diff
@@ -375,9 +398,14 @@ public void testRequiredRoutingCrudApis() throws Exception {
         }
 
         logger.info("--> verifying get with routing, should find");
-        for (int i = 0; i < 5; i++) {
-            assertThat(client().prepareGet(indexOrAlias(), "1").setRouting(routingValue).execute().actionGet().isExists(), equalTo(true));
-        }
+        assertBusy(() -> {
```

Comment:
Using assertBusy will re-execute the code block inside on assertion trip. One single assertion failure, get requests will start from 0, which is probably not needed. This is fine for code block which doesn't execute API requests. We can use assertBusy on inner assertion statement. This is applicable for this logic and other places where we asserted on entire code block.

Reply:
Got it. Will update this